src/core_plugins/timelion/public/directives/timelion_expression_input.js

```diff
@@ -158,8 +162,14 @@ app.directive('timelionExpressionInput', function ($document, $http, $interval,
           argValueSuggestions
         );
 
+        if (_.get(suggestions, 'isPrevRequestResults', false)) {
```

Comment:
I don't think this is the right way to do this (it also doesn't seem to be working well in practice - I'll post a gif shortly)

I think instead what we should do is return the input the suggestions are based off of on the suggestions object.  Then when we get here, do something like  `if (suggestions.sheet != scope.sheet) return;` along with a comment that says something like, "our sheet changed since we requested the suggestions, they are no longer valid."

I'm only saying "sheet" because that's what's on scope it seems like.

Reply:
Good idea. This is a lot simpler and easier to follow